src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -726,6 +727,13 @@ private async Task ProcessSubscriptionAsync()
 
                                     AssertCloseWhenNoDocsLeft();
 
+                                    // we might wait for new documents for a long times, lets reduce the stream capacity
+                                    if (_buffer.Capacity > MaxBufferCapacityInBytes)
+                                    {
+                                        Debug.Assert(_buffer.Length < _buffer.Capacity);
```

Comment:
don't you expect here 0?
also
it should be `Debug.Assert(_buffer.Length <= MaxBufferCapacityInBytes, $"{_buffer.Length} <= {MaxBufferCapacityInBytes}");`  ?

Reply:
added as precaution